Fix landing page for mobile screens (#24)

tolokoban opened the pull request:
This PR relates to [issue #24](https://github.com/openbraininstitute/prod-obi-site/issues/24).

It adapts the landing page layout to make it useable on screens of every size, even phones.

tolokoban committed:
Fix landing page for mobile screens (#24)

bilalesi commented on src/components/LandingPage/sections/SectionPricing/SectionPricing.tsx:
```diff
@@ -35,9 +35,9 @@ export default function SectionPricing({ className }: SectionPricingProps) {
       </p>
-      <p>
+      <div>
         <EmailButton email="support@openbraininstitute.org">Need some support?</EmailButton>
-      </p>
+      </div>
```
**nit**: it seems it need a bit of spacing on top 🤔 
<img width="1308" alt="Screenshot 2025-01-29 at 14 11 41" src="https://github.com/user-attachments/assets/1ce5eb07-88e0-48f7-8934-bd89e1871c39" />

tolokoban replied:
I let this one aside for now, because the interface will still change a lot and this is not directly related to mobile layout. But, you are right about the fact that we should add more space before the button and less between the lines inside the button.

bilalesi commented on src/components/LandingPage/CenteredColumn/CenteredColumn.tsx:
this is general comment and are nit-picks (so it's up to u @tolokoban):
1. for column style of the sections: it appear that there is double lines separator, (we can hide one)
<img width="405" alt="Screenshot 2025-01-29 at 14 16 26" src="https://github.com/user-attachments/assets/f8916775-4c2b-48f8-9650-5c09cae24edf" />


2. it might you discussed with loris about this, so ignore it 
the menu is centered, but it's different then the design


<img width="423" alt="Screenshot 2025-01-29 at 14 19 47" src="https://github.com/user-attachments/assets/2b0c9fee-5526-40c0-892d-326718584a91" />
<img width="123" alt="Screenshot 2025-01-29 at 14 20 50" src="https://github.com/user-attachments/assets/9d6a743b-9b19-4662-abdf-43d707b01645" />



3. this test in Sumusung S20, the "terms and conditions" seems centered against others

<img width="471" alt="Screenshot 2025-01-29 at 14 22 50" src="https://github.com/user-attachments/assets/eef39f1a-1605-4f61-8fa7-ed868a0c7b16" />


4. the vertical line is on top of the writing due text taking two lines (Terms & conditions)

<img width="503" alt="Screenshot 2025-01-29 at 14 24 30" src="https://github.com/user-attachments/assets/cc46448f-c8e7-4261-822b-d6717dff0c34" />

5. in different places the text seems too big for a mobile phone, 
6. for sumsung phone,  some pages is scrollable horizontally probably due some overflow,

tolokoban replied:
Thanks a lot, @bilalesi for this thorough review that helped fix many issues.
I have added my corrections here.

tolokoban committed:
Fix layout for mobiles according to review feebacks

bilalesi commented:
1. social media logos and text:

<img width="1470" alt="Screenshot 2025-01-30 at 11 14 06" src="https://github.com/user-attachments/assets/e56e4eb7-bf96-4884-9ddc-68ae511589a0" />

2. OUR MISSION: need more spaces between the title "Advance Neuroscience Understanding" and the big text
second image from design

<img width="1142" alt="Screenshot 2025-01-30 at 11 15 33" src="https://github.com/user-attachments/assets/6d0a7349-45e6-4a33-bf43-78470d9dab0b" />
<img width="171" alt="Screenshot 2025-01-30 at 11 16 30" src="https://github.com/user-attachments/assets/d8931522-4cba-4377-93db-7ee356fb3a97" />

3. PRICING: the email box should be aligned with text and have some space between them

<img width="1321" alt="Screenshot 2025-01-30 at 11 17 23" src="https://github.com/user-attachments/assets/d29342d6-8791-4612-b1c6-eddfd21d53fd" />

4. CONTACT: two emails boxes, (in the design are vertical), but here are horizantal (it might you agree with loris to change 🤔)
<img width="610" alt="Screenshot 2025-01-30 at 11 19 38" src="https://github.com/user-attachments/assets/1d7c8ad7-2c40-42de-b30b-2bc943171648" />

5. TERMS and CONDITIONS: there is tiny difference  (number and text are vertical), there is also a separator between each point (again up to u if need a change)

<img width="1191" alt="Screenshot 2025-01-30 at 11 21 31" src="https://github.com/user-attachments/assets/812c935d-bb0f-4ef0-bdd2-10f1d819a849" />

<img width="704" alt="Screenshot 2025-01-30 at 11 21 41" src="https://github.com/user-attachments/assets/e4c149a9-ea36-4257-97be-01d6d28e304f" />

6. in the mobile version 

<img width="634" alt="Screenshot 2025-01-30 at 11 25 07" src="https://github.com/user-attachments/assets/28d20c77-279a-4334-895b-7ffb10bd0d3c" />

7. mobile version (nit: the button is a bit overflowed)
<img width="520" alt="Screenshot 2025-01-30 at 11 26 17" src="https://github.com/user-attachments/assets/9aefa780-cb98-42cd-9024-5b3c1906e3b5" />

tolokoban committed:
Code review

tolokoban commented:
1. **FIXED**. This was a regression due to the fact that I removed the text when the screen was too small.
2. **FIXED**. I think this was already here with the branch I used as a base, but it's good to fix it.
3. **FIXED**.
4. This one is a personal choice based on a remark from George that he does not like wasting space. For instance, we started by stacking all the portals one upon the other, but JD and Georges decided we needed to have at least two because "there is too much empty space". So, I prefer to let this as is for now because I'm sure everything will have to be changed again soon  :-)
5. **FIXED**. It was not acceptable on mobile.
6. Idem.
7. **FIXED**. Added a max-width depending on the screen width.
